packages/eslint-plugin-internal/src/rules/plugin-test-formatting.ts

```diff
@@ -478,33 +491,99 @@ export default createRule<Options, MessageIds>({
-      AST_NODE_TYPES.ArrayExpression,
-      AST_NODE_TYPES.ObjectExpression,
-    ];
-
     return {
       // valid
       'CallExpression > ObjectExpression > Property[key.name = "valid"] > ArrayExpression':
         checkValidTest,
       // invalid - errors
-      [invalidTestsSelectorPath.join(' > ')]: checkInvalidTest,
-      // invalid - suggestions
       [[
-        ...invalidTestsSelectorPath,
-        'Property[key.name = "errors"]',
-        AST_NODE_TYPES.ArrayExpression,
+        AST_NODE_TYPES.CallExpression,
         AST_NODE_TYPES.ObjectExpression,
-        'Property[key.name = "suggestions"]',
+        'Property[key.name = "invalid"]',
         AST_NODE_TYPES.ArrayExpression,
         AST_NODE_TYPES.ObjectExpression,
       ].join(' > ')]: checkInvalidTest,
       // special case for our batchedSingleLineTests utility
       'CallExpression[callee.name = "batchedSingleLineTests"] > ObjectExpression':
         checkInvalidTest,
+
+      /**
+       * generic, type-aware handling for any old object
+       * this is a fallback to handle random variables people declare or object
+       * literals that are passed via array maps, etc
+       */
+      ObjectExpression(node): void {
+        if (checkedObjects.has(node)) {
+          return;
+        }
+
+        const type = getContextualType(
+          checker,
+          esTreeNodeToTSNodeMap.get(node),
+        );
+        if (!type) {
+          return;
+        }
+
+        const typeString = checker.typeToString(type);
+        if (/^RunTests\b/.test(typeString)) {
+          checkedObjects.add(node);
+
+          for (const prop of node.properties) {
+            if (
+              prop.type === AST_NODE_TYPES.SpreadElement ||
+              prop.computed ||
+              prop.key.type !== AST_NODE_TYPES.Identifier ||
+              prop.value.type !== AST_NODE_TYPES.ArrayExpression
+            ) {
+              continue;
+            }
+
+            switch (prop.key.name) {
+              case 'valid':
+                checkValidTest(prop.value);
+                break;
+
+              case 'invalid':
+                for (const element of prop.value.elements) {
+                  if (element.type === AST_NODE_TYPES.ObjectExpression) {
+                    checkInvalidTest(element);
+                  }
+                }
+                break;
+            }
+          }
+          return;
+        }
+
+        if (/^ValidTestCase\b/.test(typeString)) {
+          checkInvalidTest(node);
+          return;
+        }
+
+        if (/^InvalidTestCase\b/.test(typeString)) {
+          checkInvalidTest(node);
+          for (const testProp of node.properties) {
+            if (
+              testProp.type === AST_NODE_TYPES.SpreadElement ||
+              testProp.computed ||
+              testProp.key.type !== AST_NODE_TYPES.Identifier ||
+              testProp.key.name !== 'errors' ||
+              testProp.value.type !== AST_NODE_TYPES.ArrayExpression
```

Comment:
Nit: this is _almost_ the same as the previous `.type === AST_NODE_TYPES.SpreadElement || ...` check, but for the `'errors'` case. I wonder if there's a good way to dedup? 🤷 definitely not a blocker IMO.